src/App.tsx

```diff
@@ -1,24 +1,192 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  useCallback,
+  useEffect,
+  useState,
+  FC,
+  useMemo,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Filter } from './types/FilterEnum';
+import { addTodos, deleteTodos, getTodos } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoForm } from './components/TodoForm';
+import { BottomBar } from './components/BottomBar';
+import { Error } from './components/Error';
+import { ErrorMessage } from './types/ErrorEnum';
 
-const USER_ID = 0;
+const USER_ID = 10315;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterOfTodo, setFilterOfTodo] = useState(Filter.All);
+  const [hasError, setHasError] = useState(false);
+  const [idOfDeletedTodo, setIdOfDeletedTodo] = useState<number | null>(null);
+  const [
+    completedTodosID,
+    setCompletedTodosID,
+  ] = useState<number[] | null>(null);
+  const [typeOfError, setTypeOfError] = useState(ErrorMessage.EMPTY);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const loadTodoFromServer = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setHasError(true);
+      setTypeOfError(ErrorMessage.SERVER);
+    }
+  }, []);
+
+  const createTodoOnServer = useCallback(async (data: Todo) => {
+    try {
+      const todo = await addTodos(data, USER_ID);
+
+      setTodos(prevTodo => [...prevTodo, todo]);
+    } catch {
+      setHasError(true);
+      setTypeOfError(ErrorMessage.ADD);
+    }
+
+    setTempTodo(null);
+  }, []);
+
+  const deleteTodoFromServer = useCallback(async (id: number) => {
+    try {
+      const deletedTodo = await deleteTodos(id);
+
+      if (deletedTodo) {
+        setIdOfDeletedTodo(null);
+        setTodos(prevTodos => prevTodos.filter(todo => todo.id !== id));
+      }
+    } catch {
+      setHasError(true);
+      setTypeOfError(ErrorMessage.DELETE);
+    }
+  }, []);
+
+  const addTodo = useCallback((query: string) => {
+    if (!query.trim()) {
+      setHasError(true);
+      setTypeOfError(ErrorMessage.EMPTY);
+
+      return;
+    }
+
+    const data = {
+      id: 0,
+      userId: USER_ID,
+      title: query,
+      completed: false,
+    };
+
+    setTempTodo(data);
+    createTodoOnServer(data);
+  }, []);
+
+  const deleteTodo = useCallback((id: number) => {
+    setIdOfDeletedTodo(id);
+    deleteTodoFromServer(id);
+  }, []);
+
+  const deleteAllCompleted = useCallback(() => {
```

Comment:
you implement this function but i can't see `Clear completed` button)
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/91411692/3c966719-5fc4-4adc-9f8a-34609309a23f)

Reply:
it is when you focus